app/models/dialog.rb

```diff
@@ -27,27 +27,24 @@ def self.seed
     end
   end
 
-  def each_dialog_field
-    dialog_tabs.each { |dt| dt.each_dialog_field { |df| yield(df) } }
+  def each_dialog_field(&block)
+    dialog_fields.each(&block)
   end
 
   def dialog_fields
-    dialog_tabs.collect(&:dialog_fields).flatten!
+    dialog_tabs.flat_map(&:dialog_fields)
   end
 
   def field_name_exist?(name)
-    each_dialog_field { |df| return true if df.name == name }
-    false
+    dialog_fields.detect { |df| df.name == name }
```

Comment:
Should this be `.any?` instead of `.detect`, since this method is supposed to return true/false, not the actual object?